bindings/nodejs/src/functions/gpio.cc

```diff
@@ -35,6 +35,44 @@ class SolGpio : public JSHandle<SolGpio> {
     static const char *jsClassName() { return "SolGpio"; }
 };
 
+#define OPEN_GPIO(functionname, pin, jsGPIOConfig) \
+    do { \
+        sol_gpio_config config; \
+        sol_gpio *gpio = NULL; \
+\
+        sol_gpio_data *gpio_data = new sol_gpio_data; \
+        gpio_data->callback = NULL; \
+\
+        if (!c_sol_gpio_config(jsGPIOConfig, gpio_data, &config)) { \
```

Comment:
Here, you would do ```jsGPIOConfig->ToObject()``` and you wouldn't have to do it three different times in the code.